### PR TITLE
test: add `e2e` testing framework

### PR DESCRIPTION
**Pull request by aljo242**
## 1. Summary

Closes QCK-185

Add `e2e` testing framework similar to what is used in [osmosis](https://github.com/osmosis-labs/osmosis/tree/main/tests/e2e).

## 2.Type of change

- [X] New feature (non-breaking change which adds functionality)

## 3. Implementation details

As mentioned, HEAVILY based off of the [osmosis e2e framework](https://github.com/osmosis-labs/osmosis/tree/main/tests/e2e).
- utilizes `dockertest` to manage resources
- spins up 2 chains and connects them with hermes
- enables any amount of tests to be called on the existing environment

Other changes
- move `simulation` under `test/simulation` (common folder for test stuff now)
- minor refactor
- add documentation and attribution

## 4. How to test/use

Run new `make` targets:
- `make test-e2e-ci`
- `make test-e2e-debug`
- `make test-e2e-short`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,49 +4,44 @@ go 1.19
 
 require (
 	cosmossdk.io/math v1.0.0-beta.4
-	github.com/CosmWasm/wasmd v0.29.2
+	github.com/CosmWasm/wasmd v0.30.0
```
Comment (joe-bowman) [inline go.mod:7]: this change will have no impact as we `replace` this below. You'll need to check on notionals v46- wasm0.30 branch

Comment (aljo242) [reply go.mod:7]: reverted

Comment (aljo242) [reply go.mod]: https://github.com/ingenuity-build/quicksilver/pull/346/commits/dd952983ea7fa77e9c453aa3df3ee339c04e7263

File: test/e2e/configurer/chain/chain.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+		},
+		5*time.Minute,
+		time.Second,
+		"tx not received on destination chain",
+	)
+
+	c.t.Log("successfully sent IBC tokens")
+}
+
+func (c *Config) EnableSuperfluidAsset(denom string) {
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply test/e2e/configurer/chain/chain.go:183]: although i guess we might want to use the same structure to submit onboarding props

File: test/e2e/containers/config.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	// It should be uploaded to Docker Hub. QUICKSILVER_E2E_SKIP_UPGRADE should be unset
+	// for this functionality to be used.
+	previousVersionQuicksilverRepository = "quicksilverzone/quicksilver"
+	previousVersionQuicksilverTag        = "v1.4.0-rc9"
+	// Pre-upgrade repo/tag for quicksilver initialization (this should be one version below upgradeVersion)
+	previousVersionInitRepository = "quicksilverzone/quicksilver"
+	previousVersionInitTag        = "v1.2.4"
+	// Hermes repo/version for relayer
+	relayerRepository = "informalsystems/hermes"
+	relayerTag        = "1.2.0"
```
Comment (joe-bowman) [inline test/e2e/containers/config.go:34]: we're currently using v1.3.0 in prod

Comment (aljo242) [reply test/e2e/containers/config.go]: https://github.com/ingenuity-build/quicksilver/pull/346/commits/aeea4c1dce28729cc20796db446c0edf5c37693a

File: test/e2e/configurer/chain/chain.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (aljo242) [reply test/e2e/configurer/chain/chain.go]: https://github.com/ingenuity-build/quicksilver/pull/346/commits/aeea4c1dce28729cc20796db446c0edf5c37693a

**Comment by aljo242** [comment]
Workflow fails are from codecov upload issue

**Comment by aljo242** [comment]
Moving forward: 
-  define wrappers for operations
- add icq resource (one for each chain?)
- add quicksilver specific tests


I figure we can just get the framework added and verified here and then can build off of it moving forward.

**Review by joe-bowman — APPROVED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
LGTM

**Pull request merged**